src/plugins/intel_cpu/src/nodes/segment_max.cpp

```diff
@@ -0,0 +1,120 @@
+    ov::element::Type segmentIdsPrecision = getOriginalInputPrecisionAtPort(1);
+
+    if (getOriginalInputsNumber() == 2) {
+        addSupportedPrimDesc({{LayoutType::ncsp, dataPrecision}, {LayoutType::ncsp, segmentIdsPrecision}},
+                             {{LayoutType::ncsp, dataPrecision}},
+                             impl_desc_type::ref);
+    } else {
+        ov::element::Type numSegmentsPrecision = getOriginalInputPrecisionAtPort(2);
+        addSupportedPrimDesc({{LayoutType::ncsp, dataPrecision},
+                              {LayoutType::ncsp, segmentIdsPrecision},
+                              {LayoutType::ncsp, numSegmentsPrecision}},
+                             {{LayoutType::ncsp, dataPrecision}},
+                             impl_desc_type::ref);
+    }
+}
+
+bool SegmentMax::created() const {
+    return getType() == Type::SegmentMax;
+}
+
+bool SegmentMax::needPrepareParams() const {
+    return false;
+}
+
+void SegmentMax::executeDynamicImpl(const dnnl::stream& strm) {
+    execute(strm);
+}
+
+template <class T, class T_idx>
+void SegmentMax::executeImpl() {
+    const auto& data_shape = getSrcMemoryAtPort(0)->getStaticDims();
+    const auto& output_shape = getDstMemoryAtPort(0)->getShape().getStaticDims();
+    const auto empty_segment_value = fillMode == ov::op::FillMode::ZERO ? T(0) : std::numeric_limits<T>::lowest();
+    ov::reference::segment_max(getSrcDataAtPortAs<const T>(0),
+                               data_shape,
+                               getSrcDataAtPortAs<const T_idx>(1),
+                               getDstDataAtPortAs<T>(0),
+                               output_shape,
+                               empty_segment_value);
+}
+
+namespace {
+struct SegmentMaxContext {
+    SegmentMax& node;
+};
+}  // namespace
+
+template <typename T>
+struct SegmentMax::SegmentMaxExecute {
+    using TData = typename std::tuple_element<0, T>::type;
+    using TIndex = typename std::tuple_element<1, T>::type;
+    void operator()(SegmentMaxContext& ctx) {
+        ctx.node.executeImpl<TData, TIndex>();
+    }
+};
+
+void SegmentMax::execute(const dnnl::stream& strm) {
+    auto dataPrecision = getParentEdgeAt(0)->getMemory().getDesc().getPrecision();
+    auto indicesPrecision = getParentEdgeAt(1)->getMemory().getDesc().getPrecision();
+    SegmentMaxContext ctx = {*this};
+    OV_SWITCH(intel_cpu,
+              SegmentMaxExecute,
+              ctx,
+              std::tie(dataPrecision, indicesPrecision),
+              OV_CASE2(ov::element::f32, ov::element::i32, float, int32_t),
+              OV_CASE2(ov::element::f32, ov::element::i64, float, int64_t),
+              OV_CASE2(ov::element::f16, ov::element::i32, ov::float16, int32_t),
+              OV_CASE2(ov::element::f16, ov::element::i64, ov::float16, int64_t),
+              OV_CASE2(ov::element::bf16, ov::element::i32, ov::bfloat16, int32_t),
+              OV_CASE2(ov::element::bf16, ov::element::i64, ov::bfloat16, int64_t),
```

Comment:
AFAIK i64 is converted by CPU plugin to i32, could you please verify that cases for i64 are correctly executed here?

Reply:
As expected, all i64 are converted to i32 during execution. Checked by inspecting the `SegmentMax::executeImpl()` function contents during runtime.

Reply:
Are the cases for i64 are needed then?

Reply:
As discussed offline, i64 cases have been removed.